WalletWasabi/WabiSabi/Backend/Rounds/Round.cs

```diff
@@ -27,11 +28,18 @@ public Round(RoundParameters roundParameters)
 			AmountCredentialIssuerParameters = AmountCredentialIssuer.CredentialIssuerSecretKey.ComputeCredentialIssuerParameters();
 			WeightCredentialIssuerParameters = WeightCredentialIssuer.CredentialIssuerSecretKey.ComputeCredentialIssuerParameters();
 
-			Coinjoin = Transaction.Create(Network);
+			var allowedAmounts = new MoneyRange(roundParameters.MinRegistrableAmount, RoundParameters.MaxRegistrableAmount);
+			var txParams = new Parameters(roundParameters.FeeRate, allowedAmounts, allowedAmounts, roundParameters.Network);
+			CoinjoinState = new Construction(txParams);
 
 			Hash = new(HashHelpers.GenerateSha256Hash($"{Id}{MaxInputCountByAlice}{MinRegistrableAmount}{MaxRegistrableAmount}{RegistrableWeightCredentials}{AmountCredentialIssuerParameters}{WeightCredentialIssuerParameters}{FeeRate.SatoshiPerByte}"));
 		}
 
+		public State CoinjoinState { get; set; }
```

Comment:
yes, but for now since `Arena` already manipulated `Round` internals (it had no methods apart from `SetPhase`) I didn't do that yet, see also the other comment about `Alices.Add`.

Before this is merged `Round` should contain `AddAlice`, `AddBob`, `Finalize`, and `AddWitness` methods that also hide the details of `AssertConstruction()` or `AssertSigning()`, and both `Arena` and `InputRegistrationHandler` (which should probably be renamed at this point and only handle preprocessing  of `Outpoint` to `Coin`) should only use that API instead of reaching directly into `Round`'s properties.